cmd/infra/powervs/create.go

```diff
@@ -0,0 +1,1170 @@
+	PowerVSRegion          string
+	PowerVSZone            string
+	PowerVSCloudInstanceID string
+	PowerVSCloudConnection string
+	VpcRegion              string
+	Vpc                    string
+	OutputFile             string
+	Debug                  bool
+}
+
+type TimeDuration struct {
+	time.Duration
+}
+
+var unsupportedPowerVSZones = []string{"wdc06"}
+
+var powerVsDefaultUrl = func(region string) string { return fmt.Sprintf("https://%s.power-iaas.cloud.ibm.com", region) }
+var vpcDefaultUrl = func(region string) string { return fmt.Sprintf("https://%s.iaas.cloud.ibm.com/v1", region) }
+
+var log = func(name string) logr.Logger { return hypershiftLog.Log.WithName(name) }
+
+// MarshalJSON custom marshaling func for time.Duration to parse Duration into user-friendly format
+func (d *TimeDuration) MarshalJSON() (b []byte, err error) {
+	return []byte(fmt.Sprintf(`"%s"`, d.Round(time.Millisecond).String())), nil
+}
+
+// UnmarshalJSON custom unmarshalling func for time.Duration
+func (d *TimeDuration) UnmarshalJSON(b []byte) (err error) {
+	d.Duration, err = time.ParseDuration(strings.Trim(string(b), `"`))
+	return
+}
+
+type CreateStat struct {
+	Duration TimeDuration `json:"duration"`
+	Status   string       `json:"status,omitempty"`
+}
+
+type InfraCreationStat struct {
+	Vpc            CreateStat `json:"vpc"`
+	VpcSubnet      CreateStat `json:"vpcSubnet"`
+	CloudInstance  CreateStat `json:"cloudInstance"`
+	DhcpService    CreateStat `json:"dhcpService"`
+	CloudConnState CreateStat `json:"cloudConnState"`
+}
+
+// Infra resource info in IBM Cloud for setting up hypershift nodepool
+type Infra struct {
+	ID                       string            `json:"id"`
+	AccountID                string            `json:"accountID"`
+	CisCrn                   string            `json:"cisCrn"`
+	CisDomainID              string            `json:"cisDomainID"`
+	ResourceGroupID          string            `json:"resourceGroupID"`
+	PowerVSCloudInstanceID   string            `json:"powerVSCloudInstanceID"`
+	PowerVSDhcpSubnetID      string            `json:"powerVSDhcpSubnetID"`
+	PowerVSDhcpID            string            `json:"powerVSDhcpID"`
+	PowerVSCloudConnectionID string            `json:"powerVSCloudConnectionID"`
+	VpcName                  string            `json:"vpcName"`
+	VpcID                    string            `json:"vpcID"`
+	VpcCrn                   string            `json:"vpcCrn"`
+	VpcRoutingTableID        string            `json:"-"`
+	VpcSubnetName            string            `json:"vpcSubnetName"`
+	VpcSubnetID              string            `json:"vpcSubnetID"`
+	Stats                    InfraCreationStat `json:"stats"`
+}
+
+func NewCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "powervs",
+		Short:        "Creates PowerVS infrastructure resources for a cluster",
+		SilenceUsage: true,
+	}
+
+	opts := CreateInfraOptions{
+		PowerVSRegion: "us-south",
+		PowerVSZone:   "us-south",
```

Comment:
are region and zone the same?

Reply:
In us-south region it is same, other regions will have a list of zones